File: Winter2022/examples/java/Lecture 6/NetworkingURL/app/src/main/java/mobiledev/unb/ca/networkingurl/RetainedFragment.java
package mobiledev.unb.ca.networkingurl;

import android.app.Fragment;
import android.content.Context;
import android.os.Bundle;

public class RetainedFragment extends Fragment {
    static final String TAG = "RetainedFragment";
    private OnFragmentInteractionListener mListener;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setRetainInstance(true);
    }

    void onButtonPressed() {
        new HttpGetTask(this).execute();
    }

    void onDownloadFinished(String result) {
        if (null != mListener) {
            mListener.onDownloadFinished(result);
        }
    }

    @Override
    public void onAttach(Context context) {
        super.onAttach(context);
        if (context instanceof OnFragmentInteractionListener) {
            mListener = (OnFragmentInteractionListener) context;
        } else {
            throw new RuntimeException(context
                    + " must implement OnFragmentInteractionListener");
        }
    }

    @Override
    public void onDetach() {
        super.onDetach();
        mListener = null;
    }

    public interface OnFragmentInteractionListener {
        void onDownloadFinished(String result);
    }
}

File: Winter2022/examples/kotlin/Lecture 6/NetworkingURL/app/src/main/java/mobiledev/unb/ca/networkingurl/RetainedFragment.kt
package mobiledev.unb.ca.networkingurl

import androidx.fragment.app.Fragment
import android.content.Context
import java.lang.RuntimeException

class RetainedFragment : Fragment() {
    private var mListener: OnFragmentInteractionListener? = null

    fun onButtonPressed() {
        HttpGetTask(this).execute()
    }

    fun onDownloadFinished(result: String?) {
        if (null != mListener) {
            mListener!!.onDownloadFinished(result)
        }
    }

    override fun onAttach(context: Context) {
        super.onAttach(context)
        mListener = if (context is OnFragmentInteractionListener) {
            context
        } else {
            throw RuntimeException(context.toString()
                    + " must implement OnFragmentInteractionListener")
        }
    }

    override fun onDetach() {
        super.onDetach()
        mListener = null
    }

    interface OnFragmentInteractionListener {
        fun onDownloadFinished(result: String?)
    }

    companion object {
        const val TAG = "RetainedFragment"
    }
}